client/state/receipts/assembler.js

```diff
@@ -77,6 +77,9 @@
  * @returns {ReceiptData} The formatted receipt data
  */
 export function createReceiptObject( data ) {
+	const failedPurchases = Array.isArray( data.failed_purchases )
+		? data.failed_purchases
+		: Object.values( data.failed_purchases || {} );
```

Comment:
Unfortunately, this needs further refinement - it returns an array of value, however the values themselves are arrays as well. So `purchase` in the closure below is an array of failed purchases. Might need to flatten the structure.

This is the raw response from the server:

```
"failed_purchases": {
			"99265812": [{
				"product_id": 1337,
				"product_name": "Domain Transfer",
				"product_slug": "domain_transfer",
				"product_meta": "example.org",
				"product_cost": 130.5
			}]
		}
```